great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -1050,8 +1051,28 @@ def get_or_create_expectation_suite(
     return expectation_suite
 
 
-def sanitize_parameter_name(name: str) -> str:
+def sanitize_parameter_name(
+    name: str,
+    suffix: Optional[str] = None,
+) -> str:
     """
-    This method provides display-friendly version of "name" argument.
+    This method provides display-friendly version of "name" argument (with optional "suffix" argument).
+
+    In most situations, "suffix" is not needed.  However, in certain use cases, "name" argument is same among different
+    calling structures, whereby "name" can be disambiguated by addition of distinguishing "suffix" argument.  Using
+    list of "MetricConfiguration" objects as example, "metric_name" and "metric_domain_kwargs" are commoonly same among
+    them, while "metric_value_kwargs" are different (i.e., calculating values of same metric with different parameters).
+    In this case, supplying "MetricConfiguration.metric_domain_kwargs_id" as "suffix" makes sanitized names unique.
```

Comment:
In what case would we not want this behavior? Are we not doing this by default just because the hash is "ugly"? Are there other use cases for what you would want to pass into `suffix`? If not, why not just make `suffix` boolean?

Reply:
@NathanFarmer Yes -- the former, it can be ugly; in many cases we may not care.  If we want to make it permanent, then we face two alternatives: either the caller computes `name + suffix`, or we pass `MetricConfiguration`.  I felt that either was non-ideal.  The former places too much burden on caller, while the latter requires an object that may not be available.  Hence, the compromise.  Thanks.

Reply:
I see, thank you!